### DISCUSSION TITLE
New Icon Set: "DarkGrey"

### DISCUSSION THREAD

**Discussion by chris1111** [dakanji/RefindPlus#198] [General]
I created a Icon-Set-Dark Grey for RefindPlus-TOPFix
I hope you like it and RefindPlus users will like it too
[My Repos](https://github.com/chris1111/Icon-Set-DarkGrey-Refind)
Download Icons Set ➤ [refind-Icon-Set-DarkGrey.zip](https://github.com/chris1111/Icon-Set-DarkGrey-Refind/raw/refs/heads/main/refind-Icon-Set-DarkGrey.zip)

![screenshot_000](https://github.com/user-attachments/assets/f0ec40d9-c35b-4716-998c-d5dab52664f5)

I am working on another style Round Icons

![os_linux](https://github.com/user-attachments/assets/d0535dfd-25b0-44db-9cc5-30535b365f75)

**Comment by dakanji** [comment]
Nice. Very well done.

Note that RefindPlus currently has two additional "OS" icons (first row) over upstream rEFInd: `os_opencore` and `os_ventoy`.
For the "TOOL" icons (second row), there is currently an additional `tool_clean_nvram`.

BTW, the project is simply `RefindPlus`. "GOPFix" is just the name of the development branch in git as the project started with an attempt to apply fixes to GOP in rEFind.
Also, your work would get better adoption if you describe the icons as being for `rEFInd` (far more people use this).
Would still work for RefindPlus and be fully compatible if the extra ones are added.

Only small comment is that lighter elements in some of the icons, such as the Linux one, could be a smidgen brighter. These need a tiny bit more overall contrast to improve the separation from the background. A license is also required.

**Comment by chris1111** [comment]
Thanks Its [done](https://github.com/chris1111/Icon-Set-DarkGrey-Refind) 
how can i hide this multiple icon

![ScreenShot_000 copy](https://github.com/user-attachments/assets/90888978-103b-4aba-934d-e2efa71126c7)

Now my os_opencore is showing
![ScreenShot_001](https://github.com/user-attachments/assets/bd321304-bc9e-4bf2-9918-a988126e8f87)

and the linux is more brighter
![ScreenShot_000](https://github.com/user-attachments/assets/22d093e2-88b2-4433-9b8a-7366fdce2d0c)

**Comment by dakanji** [comment]
> how can i hide this multiple icon

That is the os_uefi icon. It is used for unidentified uefi binaries.

Your banner shows "rEFInd" but if you are actually runnig RefindPlus, you can try setting the `decline_help_icon` token to show a specific icon if present. You can read up this on the front page readme file. Search for `Loader Icons`.

If they are totally unwanted icons, you can either add them to an appropriate `dont_scan_XYZ` config option or simply highlight them on the screen and click on the Minus or Delete keys to hide them. This works in both rEFInd and RefindPlus.
  
BTW, you can use the `screen_rgb` config setting in RefindPlus to set any background colour you want:

The front page readme on this repo lists and summarises the additional options available.
You can find a bit more detail in the sample config file (listed on the repo front page)
Again, assuming you are on RefindPlus and not rEFInd.

> my os_opencore is showing

There is a bug with that where it can only be used in a manual stanza. Will be fixed in the next release.

**Reply by chris1111** [reply]
Of course I use RefindPlus
Everything is correct now with your recommendation. I also created another background [banner-plus.png](https://github.com/chris1111/Icon-Set-DarkGrey-Refind/blob/main/icons/banner-plus.png)
Thank you for your great work

![ScreenShot_003](https://github.com/user-attachments/assets/82279fa5-fbcd-4f62-9bfa-fc731cf04674)

**Comment by dakanji** [comment]
Well RefindPlus does not need a background banner file to show a black background colour or any other colour for that matter.

You simply do not define a banner in the config so that the internal default is used and the colour for that can be set using `screen_rgb`.

As a minimum benefit, `RefindPlus` will be written correctly. Every time I see people insist on writing it in some weird way, `rEFIndPlus`, `ReFind Plus`, `rEFInd+` etc, I regret not changing the name as I had wanted to do some time ago and start thinking maybe I should.

**Reply by chris1111** [reply]
I'm sorry for not writing the right name on my background, I'll correct it be sure but I insist on using a background because I've been [creating themes for OpenCore](https://github.com/chris1111/My-Simple-OC-Themes) for several years now and I like to use a personal Background when I like it, it's personal to each person.

**Comment by dakanji** [comment]
Sure whether to use a banner or not and what is on the banner is a choice. That is, you don't have to change the banner.

Was just pointing out that there is a option not to use one since it is a generally plain replication of the default. You do need a file to change the colour in rEFInd but in RefindPlus, the background colour can be changed as a config setting. Separate file is only really needed for elaborate graphics but can be used however wanted of course. Small speed penalty.

You can keep your banner file by all means if you like it as it is and do not have to change it.

Enjoy!

PS: You don't need to quote entire post in replies.

**Comment by chris1111** [comment]
I update the [banner](https://github.com/chris1111/Icon-Set-DarkGrey-Refind/blob/main/Icon-Set-DarkGrey/banner-plus.png) of my theme with the orig name, this is new Background
If you want a copy of the theme and maybe include it somewhere in your repos it's up to you
Attaching here
[Icon-Set-DarkGrey.zip](https://github.com/user-attachments/files/17961347/Icon-Set-DarkGrey.zip)

thanks
![ScreenShot_001](https://github.com/user-attachments/assets/64a0cb09-7767-4d67-b38b-6ca83fecb246)

**Comment by dakanji** [comment]
Currently only have the icons that come with the rEFInd project. They are not the prettiest but convenient to just stick with those.
Users can find themes via multiple means including web searches. 

There is a gallery of themes in the rEFInd documentation: https://www.rodsbooks.com/refind/themes.html.
The maintainer can add your theme there if you send him an email.

Adding the `refind-theme` here on GitHub, as you have done, will also help with discovery.
You can also add others such as `refindplus`, `refindplus-bootmanager`, `refind`

Thanks for the contrib!

**Comment by chris1111** [comment]
I have a question about MyBootMgr.dmg 
Why all the .scpt script is read only, is there an open source for this app and all this included app

**Comment by dakanji** [comment]
It just works better for me to keep it that way at this time. I don't plan to change this at this time.

**Comment by jzrodriguez98** [comment]
@chris1111 I tried configuring the refind.conf file to load this theme but I wasn't successful. I should be making a mistake. I added to my refind.conf the line that you indicated in your repo, pointing to the theme conf file but when rebooting it's not showing the theme.

**Reply by jzrodriguez98** [reply]
@chris1111 I ended up replacing an icons folder I had with your dark grey icon folder I downloaded from your repo and it worked.

**Reply by chris1111** [reply]
> @chris1111 I ended up replacing an icons folder I had with your dark grey icon folder I downloaded from your repo and it worked.

Thanks another [beauty theme here](https://github.com/chris1111/Shadow-rEFInd)

**Comment by dakanji** [comment]
> Note that RefindPlus currently has two additional "OS" icons (first row) over upstream rEFInd: `os_opencore` and `os_ventoy`.
For the "TOOL" icons (second row), there is currently an additional `tool_clean_nvram`.

I wrote this earlier and for completeness, you might want to note that two new icons will be searched for as from the next release of RefindPlus:
- **os_windows**: Will be preferred over the current `os_win` and `os_win8` if present.
  - The new `os_windows` can simply be a copy of `os_win8`
    -  `os_win8` was originally added to represent the Windows 8 and UEFI boot when that was new.
    - Has resulted in some confusion with Windows 10 and 11.
    - Some have been adding `os_win10` and `os_win11` files but these never get used unless directly added to manual stanzas.
    - Upstream should perhaps have just stuck with `os_win` back then but that is now water under the bridge.  
  - The old icons will be kept for upstream compatibility for the foreseeable future.
- **boot_windows**: Will be preferred over the current `boot_win` if present.
  - Automatically the case once `os_windows` was preferred.
  - Can be a copy of the current `boot_win`.
  - The current `boot_win` will be kept for upstream compatibility for the foreseeable future.

You can add the two to your packages by duplicating as mentioned.
If absent, the existing old ones will be used instead for the foreseeable future.

PS: You might want to make the background of your `boot_XYZ` files transparent.

**Comment by chris1111** [comment]
My four themes have already os_windows.png icons Latest here [rEFInd-Metal-Frame](https://github.com/chris1111/rEFInd-Metal-Frame)

**Comment by dakanji** [comment]
Took a look at the themes. Nice work.

A few observations:
- More important than how you choose to write "**RefindPlus**" in your banner, is how it is written in your documentation
- The `os_windows10` and `os_windows11` files will never be recognised but I suppose they could be manually set by users as replacements
- A `boot_windows` file is needed as mentioned in my previous
  - At least will be as from the next release
- Most important of all is that a theme should stick to only recommending theme related items.
  - You have put recommendations for RefindPlus that are related to your specific setup, such as the `dont_scan_XYZ` item, in a theme conf
    - These can/will cause problems for others with different setups
    - Issues related to specific setups that are not theme specific should not be recommended as general items
    - Only the items currently recommended for rEFInd should remain.
      - These are theme related and a theme config file should stick to such
  - Basically, you should only have one theme config file 
    - The items shown for rEFInd, are correct for this and for RefindPlus
    - The extras currently shown for RefindPlus, while possibly needed on your specific setup, should be removed.

 Well done!

**Comment by chris1111** [comment]
Thank you @dakanji  I will edit the config recommandation then add boot_windows.png to all of my theme
I will also correct the name RefindPlus in readme
